### DISCUSSION TITLE
Variable habit frequency

### DISCUSSION THREAD

**Discussion by sm-g** [iSoron/uhabits#270] [Feature Requests]
Some habits have changeable repeat interval, i.e. watering of plants should be done more often in summer than in winter. After changing this parameter, app apply new rule to all history (and percentage rise down, like we missed some at past, but we did not). 

Would be cool be remember (and allow edit) repetition settings change in time.

**Comment by iSoron** [comment]
Hi @sm-g, thank you for the suggestion. While it makes sense for the app to remember changes to the habit frequency, I would expect such a feature to complicate the interface and cause quite a bit of confusion. If anyone thinks this can be done in an intuitive way, please feel free to implement a prototype, and I will consider it.

**Comment by fin0** [comment]
What do you think would create the biggest part of the confusion? The interface to manage frequencies for a list of time periods for a certain habit?
Or do you think the problem would arise more in the evaluation, e.g. the graph and everything else?

**Comment by vahidri** [comment]
@iSoron
a suggestion:
- a button next to the frequency setting inside editing pop-up, which would open a simple history of frequency changes with timestamps [could be extended to a recurrent scheme, e.g. seasonal]
- Using that history to **visualize** the variable threshold by drawing approximate dotted lines over the _bar chart_ !

**Comment by jklingen** [comment]
> Would be cool be remember (and allow edit) repetition settings

I might be wrong, but from the UI perspective, I think most of the complexity of this request is in the words "allow edit". So here's my suggestion for an MVP (which would be absolutely sufficient for me):

_Given the user has an existing habit,
when they change its frequency,
then display a confirmation dialog allowing to choose whether previous scores and streaks shall be re-calculated._

My use case: my 2x per week habit works out fine, so I decided to increase to 3x. It was a bit surprising to me to see all my previous scores drop, and many streaks to break. I expected the previous values remain frozen, with my score dropping just as of today or tomorrow.
However, I am pretty sure some people prefer it that way, especially in an opposite scenario (e.g. when switching from 3x to 2x per week), so I think it would be nice to let the user decide.

**Comment by stpou** [comment]
Aside from seasonal habits, plans are made to be adjusted over time. It would be great to do so without changing the repetition.

> _Given the user has an existing habit, when they change its frequency, then display a confirmation dialog allowing to choose whether previous scores and streaks shall be re-calculated._

This is what I'm used to personally and seems to be intuitive enough.
